app/config/ServiceConfiguration.scala

```diff
@@ -63,25 +64,57 @@ class PlayBasedServiceConfiguration @Inject()(configuration: Configuration, env:
   override def retryInterval = getRequired(configuration.getMilliseconds, "aws.sqs.retry.interval").milliseconds
 
   override def s3UrlExpirationPeriod(serviceName: String): FiniteDuration = {
-    val urlExpirationConfig =
-      s"${runMode.env}.upscan.consuming-services.${replaceInvalidJsonChars(serviceName)}.aws.s3.urlExpirationPeriod"
-    val expirationDuration: FiniteDuration =
-      getRequired(configuration.getMilliseconds, urlExpirationConfig).milliseconds
-
-    if (expirationDuration <= 7.days) {
-      expirationDuration
-    } else {
-      Logger.warn(
-        s"Expiration period for key: [$urlExpirationConfig] was: [$expirationDuration]. Using maximum value of 7 days instead.")
-      1.day
+    val configKey = configKeyForConsumingService(serviceName, S3UrlExpirationPeriod.ConfigDescriptor)
+    val maybeErrorOrExpiry = configuration.getMilliseconds(configKey).map(_.milliseconds).map(validateS3UrlExpirationPeriod(_,
```

Comment:
I think the Eithers are making the code difficult to read

Maybe having the maps on individual lines or a pattern match/ for comprehension might help